src/Receiver.cc

```diff
@@ -280,30 +283,7 @@ void
 Receiver::poll()
 {
     trySendGrants();
-}
-
-/**
- * Process any Receiver timeouts that have expired.
- *
- * This method must be called periodically to ensure timely handling of
- * expired timeouts.
- *
- * @return
- *      The rdtsc cycle time when this method should be called again.
- */
-uint64_t
-Receiver::checkTimeouts()
-{
-    uint64_t nextTimeout;
-
-    // Ping Timeout
-    nextTimeout = checkResendTimeouts();
-
-    // Message Timeout
-    uint64_t messageTimeout = checkMessageTimeouts();
-    nextTimeout = nextTimeout < messageTimeout ? nextTimeout : messageTimeout;
-
-    return nextTimeout;
+    checkTimeouts();
```

Comment:
Why do we need to call `checkTimeouts` in every polling iteration? This is problematic for my shenango integration because I am trying to move away from the polling execution model. In my upcoming patch, I will remove `[Sender|Receiver]::poll` and expose `trySend` and `trySendGrants` instead. Besides, I need the `[Sender|Receiver]::checkTimeouts` method to be public so that I can provide a `Homa::checkTimeouts` API to shenango. A shenango thread will invoke this method and sleeps until the next timeout deadline.

Reply:
`[Sender|Receiver]::checkTimeouts()` are now back in the public interface. As discussed, these will be used in a future `Transport::checkTimeouts()` call which will return the time before which the method should be called again. When implementing this new Transport method, we will need to consider how information (e.g. the number of buckets) is shared between the Transport and the Sender/Receiver. I will leave this discussion for a future PR.